plugins/module_utils/config_utils.py

```diff
@@ -435,8 +435,17 @@ def set(self, value: str, setting: str) -> None:
             _setting: Element = self._config_xml_tree.find(xpath)
 
         # If the element is present we will verify it's .text value
-        elif _setting.text in [None, "", " "]:
-            raise NotImplementedError("Currently only text settings supported")
+        elif _setting.text is None or _setting.text.strip() == "":
+            # check if setting has children
+            if (
+                len(_setting.items()) > 0
+                or str(ElementTree.tostring(_setting)).count(_setting.tag) > 1
+            ):
```

Comment:
We cannot assume the theoretical edge case. We've already encountered settings that contain a boolean value. In those cases `<tag>0</tag>`, `<tag/>`, and `<tag></tag>` are considered the same in their truthy value being both `False`. Unfortunately, we cannot tell whether a tag is intended to contain child elements or not by its form in the XML.
```suggestion
            if len(_setting.items()) > 0:
```